intellij/src/nl/tudelft/watchdog/intellij/util/WatchDogUtils.java

```diff
@@ -15,26 +15,23 @@
 import java.io.IOException;
 import java.util.HashSet;
 import java.util.Set;
+import java.util.function.Function;
 
 public class WatchDogUtils extends WatchDogUtilsBase {
 
     private static Project activeProject;
 
-    private static Set<String> isWatchDogActive = new HashSet<String>();
+    private static Set<String> isWatchDogActive = new HashSet<>();
 
     /**
      * Returns the contents of the editor.
      */
     public static String getEditorContent(final Editor editor)
-            throws ContentReaderException, IllegalArgumentException {
+            throws IllegalArgumentException {
```

Comment:
Why can't editor throw ContentReaderExceptions anymore? (I.e., why could you delete the document getting part?)

Reply:
Because `getDocument` is documented as `@NotNull` and can therefore never return `null`: https://github.com/JetBrains/intellij-community/blob/dbc38b38ed57a124543f975cddede32a1a0fc353/platform/editor-ui-api/src/com/intellij/openapi/editor/Editor.java#L32-L38